fhir-search/src/main/java/com/ibm/fhir/search/parameters/QueryParameterValue.java

```diff
@@ -112,6 +143,18 @@ public String toString() {
         outputBuilder(returnString, valueCode);
         outputBuilder(returnString, valueString);
         outputBuilder(returnString, valueDate);
+        
+        if (component != null && !component.isEmpty()) {
+            returnString.append("composite[");
+            // temporarily change the delimiter; NOT thread-safe
```

Comment:
Why is this not threadsafe?  it's in the threadlocal?

Reply:
Because, for some reason, the `delim` is a member of the class and not just a a local var.
If two threads are somehow sharing the instance, they will fight over who "owns" the delim...leading to unpredictable toString behavior.

I think its ugly, but this was already the design and so instead of changing that I just wanted to flag it.

Reply:
That said, if the class is only ever used from a threadlocal, then we shouldn't really ever see an issue with this.  I don't know if thats the case or not.

Reply:
Actually, looking at this again, I don't know why I didn't just define my own local var.  I'll do that now.
I'll leave the original `delim` member alone, but I still think thats ugly.

Reply:
done